CMakeLists.txt

```diff
@@ -25,14 +25,47 @@ endif()
 
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 
-if (NOT BUILD_SHARED_LIBS)
-	add_definitions(-DLIBIIO_STATIC=1)
+add_library(iio backend.c channel.c device.c context.c buffer.c utilities.c scan.c sort.c
+	${CMAKE_CURRENT_BINARY_DIR}/iio-config.h
+)
+set_target_properties(iio PROPERTIES
+	VERSION ${PROJECT_VERSION}
+	SOVERSION ${PROJECT_VERSION_MAJOR}
+	FRAMEWORK TRUE
+	PUBLIC_HEADER iio.h
+	C_STANDARD 99
+	C_STANDARD_REQUIRED ON
+	C_EXTENSIONS OFF
+)
+if (MSVC)
+	set_target_properties(iio PROPERTIES PREFIX lib)
 endif()
 
-option(WITH_NETWORK_BACKEND "Enable the network backend" ON)
-option(WITH_TESTS "Build the test programs" ON)
-option(WITH_EXAMPLES "Build examples" OFF)
+target_include_directories(iio PUBLIC ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR})
+
+target_compile_definitions(iio PRIVATE LIBIIO_EXPORTS)
+
+target_compile_definitions(iio PUBLIC
+	# Request Posix 2008.09
+	_POSIX_C_SOURCE=200809L __XSI_VISIBLE=500
+
+	# Per http://www.mingw.org/wiki/Use_more_recent_defined_functions
+	$<$<PLATFORM_ID:MinGW>:_WIN32_WINNT=0x600 WINVER=0x600>
+
+	#full Single Unix Standard v3 (SUSv3) conformance (the Unix API)
```

Comment:
nit: `# `?

Reply:
Probably a bad copy-paste. I'll remove it. Thanks!

Reply:
Actually that was already there - I'll remove it nonetheless.